avocado/plugins/yaml_to_mux.py

```diff
@@ -392,17 +393,27 @@ def __init__(self, root, mux_path):
         self.variants = None
         self.default_params = None
         self.mux_path = mux_path
+        self.variant_ids = self._get_variant_ids()
+
+    def _get_variant_ids(self):
+        variant_ids = []
+        for variant in MuxTree(self.root):
+            variant.sort(key=lambda x: x.path)
+            fingerprint = "-".join(_.fingerprint() for _ in variant)
+            variant_ids.append("-".join(node.name for node in variant) + '-' +
+                               sha.sha(fingerprint).hexdigest()[:4])
+        return variant_ids
 
     def __iter__(self):
-        for i, variant in enumerate(self.variants, 1):
+        for vid, variant in itertools.izip(self.variant_ids, self.variants):
             '''
```

Comment:
yep, this is a remnant to not forget the version (2) is also possible and will not be part of the final version.